sru/1851858/esm-trusty-test-2.md

```diff
@@ -0,0 +1,107 @@
+# Test case 2: Start from **pp64le** cloud-image on canonistack with u-a-t installed and esm attached, upgrade to -proposed and test architecture deltas
+
+2.
+ a. Start with a fresh Ubuntu **non-x86** instance which does have u-a-t installed. Enable esm with 'ubuntu-advantage attach'. Upgrade to u-a-t from -proposed.
+ b. Confirm that machine remains attached, esm gets **disabled**, and status remains **disabled**.
+ c. Update machine token files to represent delta in architectures from contract server. Ensure `ua refresh` retains attached, but esm status becomes **n/a**
+
+```
+export UA_CONTRACT_TOKEN=<NewContractToken>
+export ARCHIVE_URL=http://archive.ubuntu.com/ubuntu
+export ADMIN_NET_ID=
+export SSH_KEY=
+export LP_USER=
+
+echo -- BEGIN test 2a: enable esm via `ubuntu-advantage attach <TOKEN>` on typical trusty-updates cloud-images which already have -updates installed
+
+# Launch a basic trusty cloud-image that is updated to latest ubuntu-advantage-tools from -updates
+cat > update-uat-trusty.yaml <<EOF
+#cloud-config
+ssh_import_id: [$LP_USER]
+package_update: true
+package_upgrade: true
+runcmd:
+ - apt-get install -qy ubuntu-advantage-tools 
+EOF
+
+echo "Creating image list for ppc64el"
+openstack image list --property architecture=ppc64el -f json > images.json
+
+# create a floating ip, keep the IP and floating-ip UUID for cleanup
+FIP=( `openstack floating ip create ext_net -f json | jq -r '. | .floating_ip_address + " " + .id'` )
```

Comment:
why a floating ip? These are "expensive" in canonistack

Reply:
Also, I didn't have `jq` installed, so an earlier step to install it would be advisable.

Reply:
Finally, there is no `ext_net`, it's `net_external`

Reply:
Server stand is ext_net and canonistack is net_external. I've adapted the script to work with both. But now I see I don't even need the external nic attached of floating IP, so I'm dropping those.